composite/market-closure/src/checks/index.ts

```diff
@@ -0,0 +1,43 @@
+import * as schedule from './schedule'
+import * as th from './tradinghours'
+import { AdapterRequest } from '@chainlink/types'
+import { Validator } from '@chainlink/external-adapter'
+
+// We check for something and get yes/no answer
+export type Check = () => Promise<boolean>
+
+export enum CheckProvider {
+  Schedule = 'schedule',
+  TradingHours = 'tradinghours',
+}
+
+const isCheckProvider = (envVar?: string): envVar is CheckProvider =>
+  Object.values(CheckProvider).includes(envVar as any)
+
+export const getCheckProvider = (): CheckProvider | undefined => {
+  const check = process.env.CHECK_TYPE
+  return isCheckProvider(check) ? (check as CheckProvider) : undefined
+}
+
+export const getCheckImpl = (type: CheckProvider | undefined, input: AdapterRequest): Check => {
+  let validator: { error: any; validated: { input: { schedule: any; symbol: string } } }
+  switch (type) {
+    case CheckProvider.Schedule:
+      validator = new Validator(input, schedule.customParams)
+      if (validator.error) throw validator.error
+      return async () => schedule.isMarketClosed(validator.validated.input.schedule || {})
```

Comment:
This is basically an fn factory:

1. Validate the input
2. Prepare required fn params
3. Return the prepared fn

I will not bother us with this in this PR, but in the future, I think we should write a snippet of code that will make it easy for us and help us avoid bugs and boilerplate. Something in the realm of:

```javascript
import { factory as schedule } from './schedule'
const isMarketClosed = schedule.isMarketClosed(input)
const res = isMarketClosed()
```

So we can either use the `isMarketClosed` fn directly or just pass the `AdapterRequest` to the factory and get it built for us.